projects/js-toolkit/packages/js-api/editor/index.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: LGPL-3.0-or-later
+ */
+
+/**
+ * Editor Config Transformer
+ *
+ * For CKEditor 4, see API of CKEDITOR.config:
+ * https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html
+ */
+
+export interface EditorConfig {
+	[key: string]: any;
+}
+
+export interface EditorConfigTransformer {
```

Comment:
Maybe we can specify a template parameter `T` here (like in https://github.com/liferay/liferay-frontend-projects/blob/6008120d95bc52f6d15cdaf1366c49e466147eb7/projects/js-toolkit/packages/js-api/data-set/index.ts#L39) so that users are able to type their editor configurations more deeply.

Not sure, however, if that would help or would be used. I would need to see a CX in action using this API to see if it would make sense...

Reply:
I though about this and decided against it, but now I'm not that sure. The reason is that `config` [definition](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html) is massive, meaning full implementation of T would be massive. I don't think this is something we would want to maintain, and I can't imagine any customers would. We would only end up using `any` as T, so my reasoning was that `[key: string]: any;` is better than nothing. **But!** I just found that CKeditor 5 has [TS interface](https://github.com/ckeditor/ckeditor5/blob/master/packages/ckeditor5-core/src/editor/editorconfig.ts) that could be used for that purpose, so it does make sense to add it. Also, some customers may be crazy enough to implement this fot CKEditor 4 :man_shrugging:

Reply:
Definitely we shall not maintain the various `T` that could apply here, as we are not editor manufacturers. However, in case `T` existed out there as a type, then this could be leveraged by CX implementors to provide strong typing on their configurations. `any` is always at hand for those who don't want this.